src/scripts.js

```diff
@@ -101,22 +104,25 @@ function reportDailyHydration(date) {
   return { ounces: day.numOunces, average: avg };
 }
 
-function reportWeeklyHydration(date) {
-  return hydrationStats.getWeeklyHydration(date);
-}
+function reportNightlySleep(date, property) {
+  const night = sleepStats.individualsSleep.find(
+    (sleep) => sleep.date === date
+  );
+  const avg = sleepStats.calculateAvg(property);
 
-// DOM manipulation functions
+  return { date: date, value: night[property], average: avg };
+}
 
-// function displayDailyHydration(date) {
-//   const report = reportDailyHydration(date);
-//   dailyWater.innerText = `Date: ${date} Ounces: ${report.ounces} Average: ${report.average}`;
-// }
+function reportWeeklySleep(date, property) {
+  const week = sleepStats.getSevenDays(date);
+  const weeksSleep = week.map((day) => {
+    return { date: day.date, [property]: day[property] };
```

Comment:
I would also like a walkthrough on how we are getting the value `day[property]`.